include/ginkgo/c_api.h

```diff
@@ -0,0 +1,851 @@
+ */
+size_t ginkgo_executor_gpu_thread_get_num_warps(gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the warp size of this thread-based executor.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the thread-based
+ * executor
+ * @return size_t The warp size of this executor
+ */
+size_t ginkgo_executor_gpu_thread_get_warp_size(gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the major version of compute capability.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the thread-based
+ * executor
+ * @return size_t The major version of compute capability
+ */
+size_t ginkgo_executor_gpu_thread_get_major_version(gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the minor version of compute capability.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the thread-based
+ * executor
+ * @return size_t The minor version of compute capability
+ */
+size_t ginkgo_executor_gpu_thread_get_minor_version(gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the closest NUMA node.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the thread-based
+ * executor
+ * @return size_t No. of the closest NUMA node
+ */
+size_t ginkgo_executor_gpu_thread_get_closest_numa(gko_executor exec_st_ptr);
+
+// DPCPP
+
+/**
+ * @brief Create a DPCPP executor
+ *
+ * @param device_id Device id
+ * @param exec_st_ptr Raw pointer to the shared pointer of the master executor
+ * @return gko_executor Raw pointer to the shared pointer of the DPCPP executor
+ * created
+ */
+gko_executor ginkgo_executor_dpcpp_create(size_t device_id,
+                                          gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the number of devices of this DPCPP executor.
+ *
+ * @return size_t No. of devices
+ */
+size_t ginkgo_executor_dpcpp_get_num_devices();
+
+/**
+ * @brief Get the number of subgroups of this item-based executor.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the item-based
+ * executor
+ * @return size_t No. of subgroups
+ */
+size_t ginkgo_executor_gpu_item_get_max_subgroup_size(gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the number of workgroups of this item-based executor.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the item-based
+ * executor
+ * @return size_t No. of workgroups
+ */
+size_t ginkgo_executor_gpu_item_get_max_workgroup_size(
+    gko_executor exec_st_ptr);
+
+/**
+ * @brief Get the number of computing units of this item-based executor.
+ *
+ * @param exec_st_ptr Raw pointer to the shared pointer of the item-based
+ * executor
+ * @return size_t No. of computing units
+ */
+size_t ginkgo_executor_gpu_item_get_num_computing_units(
+    gko_executor exec_st_ptr);
+
+/* ----------------------------------------------------------------------
+ * Library functions for creating arrays and array operations in GINKGO
+ * ---------------------------------------------------------------------- */
+DECLARE_ARRAY_OVERLOAD(int16_t, int16_t, i16)
```

Comment:
I don't think the `int16` overload is useful. It can't be used to create matrices, or basically anything relevant. Instead, a `size_t` might be more helpful, although this is also not necessary for the functions currently available in the C-API.